packages/amplify_auth_cognito/example/lib/Widgets/UpdateUserAttributes.dart

```diff
@@ -0,0 +1,165 @@
+    valueController = TextEditingController();
+  }
+}
+
+// ignore_for_file: public_member_api_docs
+class UpdateUserAttributesWidget extends StatefulWidget {
+  UpdateUserAttributesWidget();
+
+  @override
+  _UpdateUserAttributesWidgetState createState() =>
+      _UpdateUserAttributesWidgetState();
+}
+
+class _UpdateUserAttributesWidgetState
+    extends State<UpdateUserAttributesWidget> {
+  final List<_UserAttributeController> _userAttributeControllers = [
+    _UserAttributeController(keyValue: 'name'),
+    _UserAttributeController(keyValue: 'preferred_username'),
+  ];
+
+  final _formKey = GlobalKey<FormState>();
+
+  void _showSuccess(String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(backgroundColor: Colors.green[800], content: Text(message)));
+  }
+
+  void _showInfo(String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(backgroundColor: Colors.blue[800], content: Text(message)));
+  }
+
+  void _showError(String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(backgroundColor: Colors.red[900], content: Text(message)));
+  }
+
+  void _updateAttributes() async {
+    if (_formKey.currentState.validate()) {
+      try {
+        var attributes = _userAttributeControllers
+            .map(
+              (controller) => AuthUserAttribute(
+                userAttributeKey:
+                    controller.keyController.text.replaceAll('custom:', ''),
+                value: controller.valueController.text,
+              ),
+            )
+            .toList();
+        var res =
+            await Amplify.Auth.updateUserAttributes(attributes: attributes);
+        var attributesWithConfirmation = res.entries
+            .where((element) =>
+                element.value.nextStep.updateAttributeStep ==
+                'CONFIRM_ATTRIBUTE_WITH_CODE')
+            .map((element) => element.key)
+            .toList();
+        if (attributesWithConfirmation.isNotEmpty) {
+          _showInfo(
+              'Confirmation Code sent for attributes: $attributesWithConfirmation');
+        } else {
+          _showSuccess('Attributes Updated Successfully');
+        }
+      } on AmplifyException catch (e) {
+        _showError(e.message);
+      }
+    }
+  }
+
+  void _addAttribute() {
+    setState(() {
+      _userAttributeControllers.add(_UserAttributeController());
+    });
+  }
+
+  void _removeAttribute(_UserAttributeController controller) {
+    setState(() {
+      _userAttributeControllers.remove(controller);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Update Attribute'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(8),
+        child: Form(
+          key: _formKey,
+          child: ListView(
+            children: [
+              ..._userAttributeControllers.map((element) {
```

Comment:
Nit: It might be clearer (debatable) and maybe microscopically more performant to just have
```dart
children: _userAttributeControllers.map().toList()
```
as opposed to spreading the Iterable into an list

Reply:
There are two buttons in the ListView as well on lines 151/155